src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -375,6 +374,39 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         :param inputs: the input schema for the pipeline
         :return: the inputs for the engine
         """
+        if not self.cache_support_enabled and inputs.max_tokens > 1:
+            raise ValueError(
+                "The model used for inference does not support kv cache. It is "
+                "assumed that it maps from the token sequence to predicted logits."
+                "Set `max_tokens` to 1 to support that scenario."
+            )
+
+        # If the num_generated_predictions > 1, check how many times each prompt given
+        # as an input was repeated. If just once, the number of generated predictions
+        # for that particular prompt is equal to num_generated_predictions. This is done
+        # by repeating the prompt num_generated_predictions times If that
+        # prompt was given as an input multiple times, ignore the value of
```

Comment:
what's the reasoning behind this?

Reply:
If the user is already providing the same input multiple times, the number of generated predictions seems to be implied. Otherwise, if they provide an input 2 times, but then also set num_generated_predictions to 3 for example, that would be 6 predictions. But it makes more sense to just output 2 predictions.

Reply:
Caveat is that this doesn't follow hf behaviour so can remove